addons/op-badge/userscript.js

```diff
@@ -0,0 +1,17 @@
+export default async function({ addon, console }) {
+    const posts = document.querySelectorAll(".blockpost");
+
+    const op = await fetch(`https://scratchdb.lefty.one/v3/forum/topic/posts/${window.location.href.split("/")[5]}/0?o=oldest`).then(res => res.json());
```

Comment:
What is this fetch used for? Since forum topic pages display poster usernames, there's probably a way to do this without having to fetch from ScratchDB, right?

I haven't tried this myself yet, by the way.

Reply:
That wouldn't be hard to do on the first page, but what about other pages.

Reply:
I think it would be worth it to avoid this fetch call if it's the 1st page. But it's not strictly necessary.

Reply:
Is it necessary to use a third party service however? (ScratchDB)

Reply:
There's not really an API as far as I'm aware, but you could just fetch the HTML like this:
```
let id = "499628"
let data = await (await fetch(`https://scratch.mit.edu/discuss/topic/${id}/`)).text()
let html = document.createElement("html")
html.innerHTML = data.replace("<!DOCTYPE html>", "")
let author = html.querySelector(".black.username").textContent)
```

Reply:
You should probably use DOMParser instead of creating an "html" element like that